bundles/org.openhab.binding.fineoffsetweatherstation/src/main/java/org/openhab/binding/fineoffsetweatherstation/internal/handler/FineOffsetSensorHandler.java

```diff
@@ -62,6 +63,7 @@ public void updateSensorState(@Nullable SensorDevice sensorDevice) {
         }
         if (sensorDevice == null) {
```

Comment:
In which situation will this happen?

Reply:
This case occurs when a sensor thing has been configured but it is not returned by the gateway.

Reply:
Ah, thanks. So this is probably mostly relevant when unpairing a sensor because the user is getting rid of it, or can it also happen as a temporary glitch? If device is permanently removed, consider marking the thing as **GONE**:
https://www.openhab.org/docs/concepts/things.html#status-details

Comment is not directly related to this PR, just noticed it because channels are now being updated to UNDEF, perhaps because of this comment: https://github.com/openhab/openhab-addons/pull/12731#discussion_r872973142

Reply:
> So this is probably mostly relevant when unpairing a sensor because the user is getting rid of it

exactly

> or can it also happen as a temporary glitch?

no, this case will be handled before this condition is evaluated

> If device is permanently removed, consider marking the thing as GONE

Done